### ISSUE TITLE
Smarter compression for transaction updates

### ISSUE THREAD

**Issue by joshua-spacetime** [clockworklabs/SpacetimeDB#1716]
Currently message compression happens on a per-client basis. However typically many clients will subscribe to the same queries in which case we compress the same message multiple times. Note this compression happens off of the main execution thread. Depending on the implementation, removing such redundant compression may or may not require a breaking change to the API.

Note: I'm not proposing we do this. We should determine if it's worthwhile first.

**Comment by cloutiertyler** [comment]
@Centril Can you determine if this would require a breaking change?

**Comment by gefjon** [comment]
WS protocol versioning should protect us.

**Comment by cloutiertyler** [comment]
Except for the fact that we'd have to support two versions right off the bat, which I would like to avoid.

**Comment by gefjon** [comment]
True, if we want to do this in the near future it is more relevant whether it requires a different WS message schema.
